Sell/Basket/Models/CollectionPreferenceType.swift

```diff
@@ -12,4 +12,40 @@ public enum CollectionPreferenceType: String, Codable {
     case asap = "ASAP"
     case preorder = "PREORDER"
     case checkin = "CHECKIN"
+
+    init?(apolloType: ApolloType.CollectionPreferenceType?) {
+        guard
+            let apolloType = apolloType,
+            !apolloType.isUnknownType
+        else { return nil }
+        switch apolloType {
+        case .asap:
+            self = .asap
+        case .preorder:
+            self = .preorder
+        case .checkin:
+            self = .checkin
+        case .__unknown:
+            self = .asap // never enter this case
```

Comment:
If this this case will never be entered, what's your thought on calling `return nil` to avoid confusion?